controllers/common/update_scope_config.go

```diff
@@ -0,0 +1,101 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright(c) 2024 Wind River Systems, Inc. */
+
+package common
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/wind-river/cloud-platform-deployment-manager/controllers/manager"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type StarlingxInstance interface {
+	client.Object
+	GetObjectMeta() metav1.Object
+	GetStrategyRequired() string
+	SetStrategyRequired(strategy string)
+	GetDeploymentScope() string
+	SetDeploymentScope(scope string)
+}
+
+func extractScope(config string) (string, error) {
+	var data map[string]interface{}
+	err := json.Unmarshal([]byte(config), &data)
+	if err != nil {
+		return "", err
+	}
+
+	status, ok := data["status"].(map[string]interface{})
+	if !ok {
+		return "", nil
+	}
+
+	scope, ok := status["deploymentScope"].(string)
+	if !ok {
+		return "", nil
+	}
+
+	return scope, nil
+}
+
+func GetDeploymentScope(instance StarlingxInstance) (string, error) {
+	// Set default value for deployment scope
+	scope := manager.ScopeBootstrap
+	// Set DeploymentScope from configuration
+	annotation := instance.GetObjectMeta().GetAnnotations()
+	if annotation != nil {
+		config, ok := annotation["kubectl.kubernetes.io/last-applied-configuration"]
+		if ok {
+			scope, err := extractScope(config)
+			if err == nil {
+				if scope != "" {
+					scope, err = validateDeploymentScope(scope)
+					return scope, err
+				}
+			} else {
+				return scope, err
+			}
```

Comment:
Can we avoid nested `else` block and just return `scope, err` here?

Reply:
Re-structured the method, thanks for the advice